tidy3d/components/data.py

```diff
@@ -1018,6 +1133,26 @@ def __getitem__(self, monitor_name: str) -> Union[Tidy3dDataArray, xr.Dataset]:
         monitor_data = self.monitor_data.get(monitor_name)
         if isinstance(monitor_data, MonitorData):
             return monitor_data.data
+        if isinstance(monitor_data, SpatialCollectionData):
```

Comment:
This chunk probably needs a bit more documentation at the least.  Alternatively, we could create methods / properties in `MonitorData`, `SpatialCollectionData`, `PermittivityData` that define what gets returned when an object of this type is selected by SimulationData.  For example

```python
class MonitorData(...):

    # some stuff

    @property
    sim_data_getitem(self) -> xr.DataArray:
         """What gets returned by sim_data['monitor_data_name']"""
         
         return self.data
```

and then in `SimulationData.__get_item__`:

```python
def __getitem__(self, name):
    monitor_data = self.monitor_data.get(monitor_name)
    if monitor_data is None:
        raise Tidy3dKeyError("Monitor '{monitor_name}' not found in SimulationData.")
    return monitor_data.sim_data_getitm
```
what do you think?  Would just require putting each of these is instance blocks in their respective classes.